pkg/sfu/streamtracker.go

```diff
@@ -133,90 +148,175 @@ func (s *StreamTracker) Stop() {
 }
 
 func (s *StreamTracker) Reset() {
-	if s.isStopped.Load() {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	if s.isStopped {
 		return
 	}
 
+	s.resetLocked()
+}
+
+func (s *StreamTracker) resetLocked() {
 	// bump generation to trigger exit of current worker
 	s.generation.Inc()
 
-	s.countSinceLast.Store(0)
+	s.countSinceLast = 0
 	s.cycleCount = 0
 
-	s.initMu.Lock()
 	s.initialized = false
-	s.initMu.Unlock()
 
-	s.statusMu.Lock()
 	s.status = StreamStatusStopped
-	s.statusMu.Unlock()
+
+	for i := 0; i < len(s.bytesForBitrate); i++ {
+		s.bytesForBitrate[i] = 0
+	}
+	for i := 0; i < len(s.bitrate); i++ {
+		s.bitrate[i] = 0
+	}
 }
 
 func (s *StreamTracker) SetPaused(paused bool) {
-	s.paused.Store(paused)
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	s.paused = paused
+
+	if !paused {
```

Comment:
Let `pause` declare the layers inactive and resetting coming out of pause (similar behaviour to dynacast when we expect a layer to start up) for a quick start. Keep things cleaner.